STCC-117 Refactor ScratchConverter Parts to Recyclerviews

AntiDog opened the pull request.

catrobatjenkins commented:
Can a catrobat member please verify this patch? If it doesn't look evil comment with "test this please" to start the testrun.

catrobatjenkins commented:
Can a catrobat member please verify this patch? If it doesn't look evil comment with "test this please" to start the testrun.

catrobatjenkins commented:
Can a catrobat member please verify this patch? If it doesn't look evil comment with "test this please" to start the testrun.

catrobatjenkins commented:
Can a catrobat member please verify this patch? If it doesn't look evil comment with "test this please" to start the testrun.

thmq commented:
test this please
please run espresso tests

AntiDog commented:
fixed the lint issues

thmq commented:
please run espresso tests

thmq commented:
test this please

thmq commented:
tests look good 😉 I'll have a look at the code later today.

thmq commented on catroid/src/main/java/org/catrobat/catroid/ui/ScratchConverterActivity.java:
```diff
@@ -235,28 +227,13 @@ public boolean onCreateOptionsMenu(Menu menu) {
 
 	private void handleShowDetails(boolean showDetails, MenuItem item) {
-		searchProjectsListFragment.setShowDetails(showDetails);
+		getSearchFragment().setShowDetails(showDetails);
```
does this really have to be a method call? it is two lines of code and you're handing through the menu item, which seems strange. don't you think the item's text could be set in the onPrepareOptionsMenu() method? would make more sense to me, as it is exactly what you'd expect from the method and an additional call is kind of unnecessary.

thmq commented on catroid/src/main/java/org/catrobat/catroid/ui/ScratchConverterActivity.java:
```diff
@@ -272,7 +249,7 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
 			List<String> results = data.getStringArrayListExtra(
 					RecognizerIntent.EXTRA_RESULTS);
 			String spokenText = results.get(0);
```
just call it directly as data.getStringArrayListExtra(...).get(0) no need to create a list to buffer the return value.

thmq commented on catroid/src/main/java/org/catrobat/catroid/ui/ScratchConverterActivity.java:
```diff
@@ -288,22 +265,76 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
+		catrobatProgramName = catrobatProgramName == null ? job.getTitle() : catrobatProgramName;
+
+		if (job.getDownloadState() == Job.DownloadState.DOWNLOADING) {
+			AlertDialog.Builder builder = new CustomAlertDialogBuilder(this);
```
Don't use the CustomAlertDialogBuilder. I should have told you before, but it is going to be removed (with #2729), please just change it to AlertDialog.Builder.

thmq commented on catroid/src/main/java/org/catrobat/catroid/ui/ScratchConverterActivity.java:
```diff
@@ -288,22 +265,76 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
+		}
+
+		if (job.getDownloadState() == Job.DownloadState.NOT_READY || job.getDownloadState() == Job.DownloadState.CANCELED) {
+			AlertDialog.Builder builder = new CustomAlertDialogBuilder(this);
```
same here

thmq commented on catroid/src/main/java/org/catrobat/catroid/ui/ScratchConverterActivity.java:
```diff
@@ -288,22 +265,76 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
+		}
+
+		if (!StorageHandler.getInstance().projectExists(catrobatProgramName)) {
+			AlertDialog.Builder builder = new CustomAlertDialogBuilder(this);
```
here too.

thmq commented on catroid/src/main/java/org/catrobat/catroid/ui/ScratchProgramDetailsActivity.java:
```diff
@@ -71,37 +72,28 @@
+		JobViewListener, Client.DownloadCallback,
+		RVAdapter.OnItemClickListener<ScratchProgramData> {
+
+	public void onItemClick(ScratchProgramData item) {
```
this methods should be below the members. please move them.

thmq commented on catroid/src/main/java/org/catrobat/catroid/ui/ScratchProgramDetailsActivity.java:
```diff
@@ -122,26 +114,6 @@ protected void onCreate(Bundle savedInstanceState) {
 		programData = getIntent().getParcelableExtra(Constants.INTENT_SCRATCH_PROGRAM_DATA);
```
same thing as in the other activity, you set up the actionbar twice.

thmq commented on catroid/src/main/java/org/catrobat/catroid/ui/ScratchProgramDetailsActivity.java:
```diff
@@ -154,7 +126,7 @@ protected void onCreate(Bundle savedInstanceState) {
 		conversionManager.addGlobalDownloadCallback(this);
 
 		final Activity activity = this;
```
no.

thmq commented on catroid/src/main/java/org/catrobat/catroid/ui/ScratchProgramDetailsActivity.java:
```diff
@@ -154,7 +126,7 @@ protected void onCreate(Bundle savedInstanceState) {
+		findViewById(R.id.scratch_project_convert_button).setOnClickListener(new View.OnClickListener() {
 			@Override
 			public void onClick(View v) {
 				final int numberOfJobsInProgress = conversionManager.getNumberOfJobsInProgress();
```
if you need a view's context, e.g. for displaying a toast use the view's context -> view.getContext() 😉

thmq commented on catroid/src/main/java/org/catrobat/catroid/ui/ScratchProgramDetailsActivity.java:
```diff
@@ -187,16 +157,18 @@ protected void onStart() {
 	private void setUpActionBar() {
-		getSupportActionBar().setTitle(R.string.title_activity_scratch_converter);
-		getSupportActionBar().setHomeButtonEnabled(true);
+		setSupportActionBar((Toolbar) findViewById(R.id.toolbar));
```
https://stackoverflow.com/questions/8999591/difference-between-getstring-and-getresources-getstring/8999657

AntiDog commented:
@thmq are you done reviewing it? I already made the above-suggested changes, should I push them or wait till you have reviewed all of it?

thmq commented:
please push them and rebase

AntiDog commented:
Done

AntiDog commented:
Done

thmq commented on catroid/src/main/java/org/catrobat/catroid/ui/ScratchConverterActivity.java:
```diff
@@ -113,18 +120,20 @@ public void onCreate(Bundle savedInstanceState) {
+	@Override
+	protected void onPostCreate(Bundle savedBundle) {
+		super.onPostCreate(savedBundle);
+		getSlidingUpFragment().getFinishedFailedJobsAdapter().setScratchProgramOnClickListener(this);
```
do you have an answer here?

AntiDog replied:
TBH I'm not 100% sure yet myself if it has to be done this way. If we do it at the end of onCreate we crash with a null pointer exception because the finishedJobsAdapter is not yet initialized because that happens within onActivityCreated() of ScratchConverterSlidingUpPanelFragment which gets called after onCreate of ScratchConverterActivity. Do you have a better solution in mind here?

AntiDog commented on catroid/src/main/java/org/catrobat/catroid/ui/ScratchConverterActivity.java:
```diff
@@ -212,52 +196,34 @@ public void showSlideUpPanelBar(final long delayMillis) {
+		getSearchFragment().setShowDetails(showDetails);
+		int titleId = showDetails ? R.string.hide_details : R.string.show_details;
+		menu.findItem(R.id.show_details).setTitle(titleId);
+		return true;
```
Thanks for pointing this out! You are right we don't need it at all... Seems like we overlooked this one and did it the way it was done before it was a RecyclerViewFragment. I removed it now...

thmq commented:
test this please

thmq commented:
please run espresso tests

thmq commented:
test this please

thmq commented:
please run espresso tests

cheidenreich commented:
test this please

cheidenreich commented:
please run espresso tests

thmq commented:
@AntiDog this does not even build. You should have a look at it, rebasing alone does not magically fix everything and you should at least test if it works locally before you push and run it on jenkins.

AntiDog commented:
@thmq  I am very very sorry about that. I have no idea why it was this way, all I changed between the testing and pushing was squashing the commits I made... Probably I did some mistake there... I fixed it now.

thmq commented:
@AntiDog don't worry about it 😉 I just want you to know that it is important to test the builds locally because resolving the merge conflicts does not necessary mean that everything works as intended. As discussed yesterday, you should actively work on your prs and not just open them and wait for us to do  the rest.

thmq commented:
test this please

thmq commented:
please run espresso tests

thmq commented on catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java:
```diff
@@ -191,16 +188,7 @@ public void onPause() {
 	@Override
 	public boolean onCreateOptionsMenu(Menu menu) {
 		getMenuInflater().inflate(R.menu.menu_main_menu, menu);
-
```
I know I told you to remove this but can you please reset the MainMenuActivity to the develop's state? we still need that.

AntiDog replied:
ok changed this + the colors.xml is now also on develop state, previously it didn't have the beta_color.
